experimental/dds/tree2/src/test/shared-tree/fuzz/fuzzEditReducers.ts

```diff
@@ -100,10 +107,46 @@ function applySequenceFieldEdit(tree: ISharedTree, change: FuzzFieldChange): voi
 	}
 }
 
+function applyValueFieldEdit(tree: ISharedTree, change: FuzzDelete): void {
+	const field = tree.editor.sequenceField(
+		change.firstNode?.parent,
+		change.firstNode?.parentField,
+	);
+	field.delete(change.firstNode?.parentIndex, change.count);
+}
+
+function applyOptionalFieldEdit(tree: ISharedTree, change: FuzzFieldChange): void {
+	switch (change.type) {
+		case "insert": {
+			const field = tree.editor.optionalField(change.parent, change.field);
+			field.set(singleTextCursor({ type: brand("Test"), value: change.value }), false);
+			break;
+		}
+		case "delete": {
+			const field = tree.editor.optionalField(
+				change.firstNode?.parent,
+				change.firstNode?.parentField,
+			);
+			field.set(undefined, true);
+			break;
+		}
+		default:
+			fail("Invalid edit.");
+	}
+}
+
 function applyNodeEdit(tree: ISharedTree, change: FuzzNodeEditChange): void {
-	switch (change.nodeEditType) {
-		case "setPayload": {
-			tree.editor.setValue(change.path, change.value);
+	switch (change.type) {
+		case "sequence": {
```

Comment:
May as well do 

```ts
case "sequence":
case "value":
case "optional:
	tree.editor.setValue(change.edit.path, change.edit.value);
	break;
default:
	fail("Invalid Edit");
```

Reply:
merged the tree cases to reuse same case block